core/src/lib/mem_pool.cc

```diff
@@ -194,23 +192,19 @@ int32_t SizeofPoolMemory(POOLMEM* obuf)
 /* Realloc pool memory buffer */
 POOLMEM* ReallocPoolMemory(POOLMEM* obuf, int32_t size)
 {
-  char* cp = (char*)obuf;
-  void* buf;
-  int pool;
-
   ASSERT(obuf);
-  P(mutex);
-  cp -= HEAD_SIZE;
-  buf = realloc(cp, size + HEAD_SIZE);
+  char* cp = (char*)obuf - HEAD_SIZE;
+
+  void* buf = realloc(cp, size + HEAD_SIZE);
```

Comment:
if `cp` is not used otherwise, we could move `(char*)obuf - HEAD_SIZE` into the `realloc()` call.
```suggestion
  void* buf = realloc((char*)obuf - HEAD_SIZE, size + HEAD_SIZE);
```